src/pm/hydra/tools/bootstrap/external/slurm_query_node_list.c

```diff
@@ -92,17 +92,17 @@ static HYD_status list_to_nodes(char *str)
     /* compile element regex for old format: "h14" */
     regcomp(&ematch_old, "([a-zA-Z]+[0-9]+)", REG_EXTENDED);
 
-    /* compile group-0 regex for new format: "h-[00-12,14] | h-14" */
-    regcomp(&gmatch_new[0], "(,|^)([a-zA-Z]+[0-9]*-)(\\[[-,0-9]+\\]|[0-9]+)(,|$)", REG_EXTENDED);
+    /* compile group-0 regex for new format: "h00-[00-12,14] | h00[00-12,14] | h00-14 | h0014" */
+    regcomp(&gmatch_new[0], "(,|^)([a-zA-Z0-9]+-?)(\\[[-,0-9]+\\]|[0-9]*)(,|$)", REG_EXTENDED);
 
-    /* compile group-1 regex for new format: "00-12 | 14" */
-    regcomp(&gmatch_new[1], "([[,]|^)([0-9]+-[0-9]+|[0-9]+)([],]|$)", REG_EXTENDED);
+    /* compile group-1 regex for new format: "00-12 | 14 | " */
+    regcomp(&gmatch_new[1], "([[,]|^)([0-9]+-[0-9]+|[0-9]*)([],]|$)", REG_EXTENDED);
 
     /* compile range regex for new format: "00-12" */
     regcomp(&rmatch_new, "([0-9]+)-([0-9]+)", REG_EXTENDED);
 
-    /* compile element regex for new format: "14" */
-    regcomp(&ematch_new, "([0-9]+)", REG_EXTENDED);
+    /* compile element regex for new format: "14 | " */
+    regcomp(&ematch_new, "([0-9]*)", REG_EXTENDED);
```

Comment:
Should this really be a `*` instead of a `+`?  I thought `+` was correct.

Reply:
Well, since now hostnames can also have numbers, we need to make sure that we still enter the inner while loops matching either `ranges` or single numerical identifiers. For example, now we can have hostnames of the type: `host0001` where `basename` is `host00` and numerical identifier is `01`. The first while loop will match `host0001` and not `host00` as basename. Thus following matches won't find any numerical identifier and will fail to detect the right hostname, unless in these cases we also match the null string. Does this make sense?

Reply:
It does.  Thanks.

Reply:
Do you want me to sign this for you and merge it?

Reply:
Yes, thanks.